### PR TITLE
[BotBuilder] Migrate CloudAdapterBase

### PR DESCRIPTION
**Pull request by Batta32**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+        return connectorClient.getConversations()
+            .deleteActivity(reference.getConversation().getId(), reference.getActivityId());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public CompletableFuture<Void> continueConversation(
+        String botAppId,
```
Comment (matiasroldan6) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java:171]: I suppose there's a good reason but why this is a string and not a claimsIdentity?

Comment (matiasroldan6) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java:171]: my bad, confused with the other overload

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+        ClaimsIdentity claimsIdentity,
+        String oauthScope,
+        ConnectorClient connectorClient,
+        RestOAuthClient userTokenClient,
+        BotCallbackHandler callback,
+        ConnectorFactory connectorFactory
+    ) {
+        TurnContext turnContext = new TurnContextImpl(this, activity);
+        turnContext.getTurnState().add(BotAdapter.BOT_IDENTITY_KEY, claimsIdentity);
+        turnContext.getTurnState().add(BotFrameworkAdapter.CONNECTOR_CLIENT_KEY, connectorClient);
```
Comment (matiasroldan6) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java:412]: Strange but I suppose there's good reason for this (the key I mean)

Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java:412]: Yes, this is something that can confirm during the validation. The `CloudAdapterBase` is also getting the `connectorClient` using that key, but in C# it's never assigned.

I added that key but we need to confirm this

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+                } else if (activity.isType(ActivityTypes.INVOKE_RESPONSE)) {
+                    context.getTurnState().add(BotFrameworkAdapter.INVOKE_RESPONSE_KEY, activity);
+                    response = null;
+                } else if (
+                    activity.isType(ActivityTypes.TRACE)
+                        && !StringUtils.equals(activity.getChannelId(), Channels.EMULATOR)
+                ) {
+                    // no-op
+                    response = null;
+                } else if (StringUtils.isNotBlank(activity.getReplyToId())) {
```
Comment (matiasroldan6) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java:107]: I see you simplified an internal if and else here with an else if and then another else. I guess you did it for simplification. I understand it's ok 🤔

Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java:107]: Yes, also, I took the same implementation from [BotFrameworkAdapter](https://github.com/microsoft/botbuilder-java/blob/main/libraries/bot-builder/src/main/java/com/microsoft/bot/builder/BotFrameworkAdapter.java#L564) as it was the same code

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+
+        if (activities.size() == 0) {
+            return Async.completeExceptionally(
+                new IllegalArgumentException("Expecting one or more activities, but the array was empty.")
+            );
+        }
+
+        logger.info(String.format("sendActivities for %d activities.", activities.size()));
+
+        return CompletableFuture.supplyAsync(() -> {
```
Comment (matiasroldan6) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java:79]: Never seen this supplyAsync thing. Not sure what this is for. It's not necessarily related to the approval of this PR though

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+                Activity activity = activities.get(index);
+
+                activity.setId(null);
+                ResourceResponse response;
+
+                logger.info(String.format("Sending activity. ReplyToId: %s", activity.getReplyToId()));
+
+                if (activity.isType(ActivityTypes.DELAY)) {
+                    int delayMs = (int) activity.getValue();
+                    try {
```
Comment (matiasroldan6) [inline libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java:92]: this try/catch is not in C#. Guess it's necessary for the Thread.sleep

Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java:92]: Exactly

**Review by matiasroldan6 — APPROVED** [review]

File: libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java (hunk 4 above)
Comment (Batta32) [reply libraries/bot-builder/src/main/java/com/microsoft/bot/builder/CloudAdapterBase.java:79]: I took the same implementation from BotFrameworkAdapter as it was the same code. However, if i'm not wrong, it's very similar to the `thenCompose`

**Pull request merged**